pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -305,6 +315,11 @@ func (r *ReconcileClusterDeployment) Reconcile(request reconcile.Request) (recon
 	} else if err != nil {
 		cdLog.WithError(err).Error("error looking for install job")
 		return reconcile.Result{}, err
+	} else {
```

Comment:
No need for the else given the return above.

Reply:
what if it goes inside if err != nil && errors.IsNotFound(err)  statement on line 312 and sets existingJob to nil? Now if we don't have the else and if we only have the if existingJob.Status.Succeeded > 0 && !cd.Status.Installed {
			reportDurationMetricsFlag = true
} statement, wont that cause an error when it tries to resolve existingJob.Status.Succeeded since existingJob is nil?

Reply:
My mistake you are correct.